File: android/src/main/java/com/github/quadflask/react/navermap/OverlayImages.java
package com.github.quadflask.react.navermap;

import com.naver.maps.map.overlay.OverlayImage;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public class OverlayImages {
    private static Map<String, OverlayImage> store = new ConcurrentHashMap<>();

    public static void put(String uri, OverlayImage image) {
        store.put(uri, image);
    }

    public static OverlayImage get(String uri) {
        final OverlayImage overlayImage = store.get(uri);
        return overlayImage;
    }
}

File: android/src/main/java/com/github/quadflask/react/navermap/util/OverlayImages.kt
package com.github.quadflask.react.navermap.util

import com.naver.maps.map.overlay.OverlayImage
import java.util.concurrent.ConcurrentHashMap

object OverlayImages {
  private val store: MutableMap<String, OverlayImage> = ConcurrentHashMap()

  fun put(uri: String, image: OverlayImage) {
    store[uri] = image
  }

  operator fun get(uri: String?): OverlayImage? {
    return store[uri]
  }
}
